### ISSUE TITLE
FMS-2023.02 release

### ISSUE THREAD

**Issue by jkbk2004** [NOAA-EMC/ufsatm#684]
## Description
1. Code changes are needed to update with the new version of FMS-2023.02. See the release notes here: https://github.com/NOAA-GFDL/FMS/releases/tag/2023.02
2.  Old fmsio has been deprecated and will only be compiled with the flag “use_deprecated_io”.
3. A GFDL branch was created:  https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/tree/deprecated_io_emc

## Changes in the above feature branch:
1. model/multi_gases.F90: optional read multi gas namelist
2. minor update to use fms2_io_mod/file_exists, fms_io_mod/file_exist in tools/fv_iau_mod.F90 and fv_restart.F90

**Comment by XiaqiongZhou-NOAA** [comment]
@jkbk2004 : Lauren's GFDL dycore branch has the following two commits related to FMS 2023.02 release:
1) [remove deprecated fms function from multi_gases · laurenchilutti/GFDL_atmos_cubed_sphere@e0f6a20 (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/e0f6a201f07560266a37203fd022561e6c80b2b5)
2) [removing remaining old fms functions · laurenchilutti/GFDL_atmos_cubed_sphere@f83302b (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/f83302b45033a067055ec5e2ab4f2f417872e821)

What do you mean about #4 about  a new time scale Rayleigh damping et al? They are already in dev/emc dynamic core if you mean about the parameter "fast_tau_w_sec" ?

**Comment by jkbk2004** [comment]
> @jkbk2004 : Lauren's GFDL dycore branch has the following two commits related to FMS 2023.02 release:
> 
> 1. [remove deprecated fms function from multi_gases · laurenchilutti/GFDL_atmos_cubed_sphere@e0f6a20 (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/e0f6a201f07560266a37203fd022561e6c80b2b5)
> 2. [removing remaining old fms functions · laurenchilutti/GFDL_atmos_cubed_sphere@f83302b (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/f83302b45033a067055ec5e2ab4f2f417872e821)
> 
> What do you mean about #4 about a new time scale Rayleigh damping et al? They are already in dev/emc dynamic core if you mean about the parameter "fast_tau_w_sec" ?

![diff_fv_arrays](https://github.com/NOAA-EMC/fv3atm/assets/38663008/6df5a676-c59c-4699-98dd-288406a2b8f9)

**Comment by jkbk2004** [comment]
> > @jkbk2004 : Lauren's GFDL dycore branch has the following two commits related to FMS 2023.02 release:
> > 
> > 1. [remove deprecated fms function from multi_gases · laurenchilutti/GFDL_atmos_cubed_sphere@e0f6a20 (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/e0f6a201f07560266a37203fd022561e6c80b2b5)
> > 2. [removing remaining old fms functions · laurenchilutti/GFDL_atmos_cubed_sphere@f83302b (github.com)](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/commit/f83302b45033a067055ec5e2ab4f2f417872e821)
> > 
> > What do you mean about #4 about a new time scale Rayleigh damping et al? They are already in dev/emc dynamic core if you mean about the parameter "fast_tau_w_sec" ?
> 
> ![diff_fv_arrays](https://user-images.githubusercontent.com/38663008/259168652-6df5a676-c59c-4699-98dd-288406a2b8f9.png)

I just diff the difference of each routine at core level. So, are the parameters pretty much rearranged?

**Comment by XiaqiongZhou-NOAA** [comment]
Which version of dev/emc are you using?
Check line 701 here:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/dev/emc/model/fv_arrays.F90

**Comment by jkbk2004** [comment]
> Which version of dev/emc are you using? Check line 701 here: https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/dev/emc/model/fv_arrays.F90

Sorry! sounds like I am pointing to wrong hash. Let me re-write the issue.

**Comment by jkbk2004** [comment]
@laurenchilutti @XiaqiongZhou-NOAA one question about line 44 of molecular_diffusion.F90 in the deprecated_io_emc branch. It contains `use fms_mod,            only: check_nml_error, open_namelist_file, close_file`. Does it need to clean up as multi_gases.F90 line 45 ?

**Comment by laurenchilutti** [comment]
> @laurenchilutti @XiaqiongZhou-NOAA one question about line 44 of molecular_diffusion.F90 in the deprecated_io_emc branch. It contains `use fms_mod, only: check_nml_error, open_namelist_file, close_file`. Does it need to clean up as multi_gases.F90 line 45 ?

Yes, the same changes from multi_gases.F90 will need to be applied to any files that open namelist files with the `open_namelist_file` and `close_file` subroutines from fms.
